### ISSUE TITLE
Wrong date-Object when parsing strings without date

### ISSUE THREAD

**Issue by metromorph-dev** [foxbunny/datetimejs#2]
Hi,
I think we discovered a bug. If you parse without using a date the result points to the last day of the previous month.
e.g.:
```
datetime.strptime("March 2019", "%B %Y") // results is 28.02.2019
datetime.strptime("2019", "%Y") // result is 31.12.2018
```
I'd expect it to give me the first day of the month and the first day of the year.

The Problem seems to be in the parse routine where you initialize the `meta` object with the default values. For `date` the default value should be `1` because `0` leads to the last day of the previous month.

https://github.com/foxbunny/datetimejs/blob/c6dc583af33efcea931e442e01b97e5a10d77571/datetime.js#L555

We've attached a [patch](https://github.com/foxbunny/datetimejs/files/3264827/0001-Added-fix-strptime-2013-Y-.getFullYear-2013.zip) with tests and a fix or if you like we can create a pull request.

Regards,
Gideon

**Comment by foxbunny** [comment]
Hi Gideon, and thanks for taking the time. Yeah, the patch looks good. And sorry you had to put up with the weird variable names. That's probably from CoffeeScript-to-JavaScript conversion which I never got around to cleaning up. 

Anyway, feel free to open a PR at your own leisure, or let me know if you'd rather that I patch that in.

**Comment by metromorph-dev** [comment]
Thank you for the quick answer and action. Yes the variable names are a bit weird, but that's ok.
Is it possible to publish a new version 0.3.5?

**Comment by foxbunny** [comment]
Yes, I'll take care of it in a few minutes.

**Comment by foxbunny** [comment]
@metromorph Published as 0.3.5. Let me know if you run into issues. It's been a while since I published this package, so might have broken something while cleaning up the `package.json`.

**Comment by metromorph-dev** [comment]
Thank you so much! It seems to work just fine. Could you please do me the favour and publish it to npm?
